### PR TITLE
UI for archiving

### PR DESCRIPTION
**Pull request by PiotrKozlowski**
### User with `book:edit` permission should be able to remove books:
image

### User with `module:edit` should be able to remove module:
image

### User with `resources:manage` should be able to remove resources:
image

### User with `editing-process:manage` should be able to remove processes:
image

### User with `role:edit` permission should be able to remove roles:
image

### Super user in elevated session should be able to remove teams:
image

### Users in support team now can close and reopen tickets:
image
image

### When ticket is closed there message input should not be visible

### CODE REVIEW

File: src/containers/Chat/components/Messages/index.tsx [hunk 1]
```diff
@@ -243,3 +244,27 @@ export const UserJoined = ({ data: { id, users } }: { data: UserJoinedType }) =>
...
+      <span>
+        <Localized
+          id="chat-state-changed"
+          $user={userName}
+          anchor={<Link to={`/users/${user}`}/>}
+          $flag={flag}
+          $value={value}
+        >
+          {/* eslint-disable-next-line react/jsx-no-useless-fragment */}
+          <></>
```
Comment (aiwenar) [inline src/containers/Chat/components/Messages/index.tsx]: Why is this required?

File: public/locale/en/ui.ftl [hunk 2]
```diff
@@ -1974,6 +2024,20 @@ chat-users-joined = New users has joined:
...
+# - $flag (number): conversation flag which was changed.
+# - $value (number): new state of given flag.
+chat-state-changed = <anchor>{ $user }</anchor> { $flag ->
+  [1] { $value ->
+    [0] reopened ticket.
+    [1] closed ticket.
+   *[unknown] made unknown action.
+  }
+ *[unknown] made unknown action.
+}
```
Comment (aiwenar) [inline public/locale/en/ui.ftl]: This should be handled with different messages.

File: src/containers/Chat/components/Messages/index.tsx (hunk 1 above)
Comment (PiotrKozlowski) [reply src/containers/Chat/components/Messages/index.tsx]: Without this `<Localized>` will output `<anchor>John doe</anchor> has joined.` instead of:
[John doe](#link) has joined.

File: src/containers/Chat/components/Messages/index.tsx [hunk 3]
```diff
@@ -243,3 +244,26 @@ export const UserJoined = ({ data: { id, users } }: { data: UserJoinedType }) =>
...
+  { data: { id, user, date, flag, value } }: { data: StateChangedType }) => {
+  const usersMap = useSelector((state: State) => state.user.users)
+  const userName = usersMap.has(user) ? usersMap.get(user)!.name : 'Undefined user'
+
+  return (
+    <div className="chat__msg chat__msg--state-changed" data-id={id}>
+      <Time time={date} />
+      <span>
+        <Localized
+          id={`chat-state-changed-flag-${flag}`}
```
Comment (aiwenar) [inline src/containers/Chat/components/Messages/index.tsx]: I didn't notice this previously, but this code won't work once we add more flags. If multiple flags were changed at once only a single event is emitted for them all, and both `flags` and `value` are bitmasks. Thus this approach would require a separate message for every possible combination of flags and values.

File: src/containers/Chat/components/Messages/index.tsx (hunk 1 above)
Comment (aiwenar) [reply src/containers/Chat/components/Messages/index.tsx]: This should be documented.

**Comment by MartaSztuka** [comment]
Tested
Screenshot from 2020-01-14 15-06-15
Screenshot from 2020-01-14 15-06-00
Screenshot from 2020-01-14 15-05-19
